src/ferui-components/tree-view/tree-view-component.ts

```diff
@@ -146,10 +149,16 @@ export class FuiTreeViewComponent<T> implements OnInit, OnDestroy {
           return this.createTreeNode(child, null);
         });
         this.scrollViewArray = this.nonRootArray;
+        if (this.autoNodeSelector && this.scrollViewArray.length > 0) {
```

Comment:
`this.scrollViewArray` might be null or undefined too (if the dev forgot to specify a treeNodeData) so your check there might be triggering a console error. 
You need to handle this case above this condition and throw an error with explanations that we need to set the treeNodeData so that `this.scrollViewArray` can't be null nor undefined etc...
Same comment for line 159.

Reply:
@JoMas971 this isn't necessarily true. For the NonRoot tree view for line 152 to even be read the developer must pass in an instance of the NonRootTreeNode and I check that in line 142. Afterwards, we call the callback method from their data retriever, and this block of code is only read IF successful (meaning we get back an array, whether empty or not) we then create the `scrollViewArray` therefore the scrollViewArray will always be defined from the success callback. 

For the regular `treeNodeData` usage if the developer does indeed pass in null, we would still wrap their null treeNodeData using the `createTreeNode` method which is the wrapper for it and make the `scrollViewArray` from this wrapper on line 158. So again the scrollViewArray won't be undefined or null.

Reply:
Ok good. Ignore my comment then.